### PR TITLE
FR-15396: add support for vue router 4 basename

### PR DESCRIPTION
**Pull request by TomerFrontegg**
Screenshot 2024-03-03 at 14 35 29

### CODE REVIEW

File: packages/vue/src/helpers.ts [hunk 1]
```diff
@@ -109,3 +109,13 @@ export const simpleMappers = (subState: string, props: any, getter: any) => func
 
   return slices.reduce((result, prop) => Object.assign({}, result, { [prop]: getter(prop)(state) }), {});
 };
+
+
+/**
+ * vue-router base name was changed in v4 from `base` to `history.base`
```
Comment (doregg) [inline packages/vue/src/helpers.ts:115]: 👑

Comment (TomerFrontegg) [reply packages/vue/src/helpers.ts:115]: ❤️

File: packages/vue/src/index.ts [hunk 2]
```diff
@@ -105,7 +105,7 @@ const Frontegg: PluginObject<PluginOptions> | any = {
     fronteggApp = initialize({
       ...rest,
       onRedirectTo,
-      basename: router?.options.base,
+      basename: getRouterBaseName(router),
```
Comment (doregg) [inline packages/vue/src/index.ts:108]: pay attention that here they used optional for router.. they do it also in line 104. maybe worth adding `?` to the helper function you created

**Review by doregg — APPROVED** [review]

File: packages/vue/src/index.ts (hunk 2 above)
Comment (TomerFrontegg) [reply packages/vue/src/index.ts:108]: I already did

**Review by eran-frontegg — APPROVED** [review]

File: packages/vue/src/helpers.ts [hunk 3]
```diff
@@ -109,3 +109,13 @@ export const simpleMappers = (subState: string, props: any, getter: any) => func
...
+ * @param router VueRouter v3 or v4
+ * @returns router base name
+ */
+export const getRouterBaseName = (router?: VueRouter): string | undefined => {
+  return router?.options.base || (router?.options as any).history?.base
```
Comment (yuvalotem1) [inline packages/vue/src/helpers.ts:120]:
```suggestion
  return router?.options.base ?? (router?.options as any).history?.base
```

Comment (TomerFrontegg) [reply packages/vue/src/helpers.ts:120]: I just extracted the logic, I don't wanna change it

**Review by yuvalotem1 — APPROVED** [review]

**Pull request merged**